README.md

```diff
@@ -27,70 +35,90 @@ This contract is deployed on mainnet at TBD
 
 ## Development
 
-### Setup
+Create a copy of `.env.example` and fill in the environment variables, the proceed to the following sections.
 
-Install dependencies:
+### Contract Setup
 
 ```sh
+# Install dependencies
 $ yarn install
-```
-
-### Compile
-
-Compile the smart contracts with Hardhat:
 
-```sh
+# Compile the smart contracts with Hardhat
 $ yarn compile
-```
 
-### TypeChain
+# Generate TypeChain artifacts
+$ yarn typechain
 
-Compile the smart contracts and generate TypeChain artifacts:
+# Lint the Solidity code
+$ yarn lint:sol
 
-```sh
-$ yarn build
-```
+# Lint the TypeScript code
+$ yarn lint:ts
 
-### Lint Solidity
+# Run tests
+$ yarn test
 
-Lint the Solidity code:
+# Generate the code coverage report
+$ yarn coverage
 
-```sh
-$ yarn lint:sol
+# Delete the smart contract artifacts, the coverage reports and the Hardhat cache
+$ yarn clean
 ```
 
-### Lint TypeScript
+### Python Setup
 
-Lint the TypeScript code:
+For the following section you'll need python. To get setup, follow the commands below:
 
 ```sh
-$ yarn lint:ts
+# Create a new virtual environment in this directory
+python3 -m venv venv
+
+# Activate the virtual environment
+source ./venv/bin/activate
+
+# Install dependencies
+pip install -r requirements.txt
 ```
 
-### Test
+### Test Workflow
+
+This section explains how to deploy a local instance of the contract and run Python scripts to
+check the contract state. The final versions of this script live in the
+[gitcoin/web](https://github.com/gitcoinco/web/) repo, but were developed here.
 
-Run the Mocha tests:
+First let's deploy the contracts locally.
 
 ```sh
-$ yarn test
+$ yarn deploy:local
 ```
 
-### Coverage
+Wait a few seconds for that to complete.
 
-Generate the code coverage report:
+Now we want to set a payout mapping and verify the results before finalizing it. We can do this with
+the below command, which will set the payout mapping and save the mapping to `outputs/payouts.json`
+so we can verify it later
 
 ```sh
-$ yarn coverage
+$ yarn set-payouts
 ```
 
-### Clean
-
-Delete the smart contract artifacts, the coverage reports and the Hardhat cache:
+Let's compare the total value of the payouts mapping from the events to what we'd expect from
+`payouts.json`, and let's compare that to the DAI balance of the contract.
 
 ```sh
-$ yarn clean
+# Make sure the virtual environment is activated
+$ source ./venv/bin/activate
```

Comment:
Can we just make this part of the `yarn verify-payouts` script to avoid errors/confusion?

Reply:
Good call! I'll update that

Reply:
Added in be86791